### ISSUE TITLE
JSON properties containing strings of valid JSON are being parsed

### ISSUE THREAD

**Issue by minznerjosh** [mikro-orm/mikro-orm#4476]
**Describe the bug**
When using PostgreSQL and an entity with a JSON property, if the property is set to a string that happens to be valid JSON, the string is being parsed when populated from the database.

**Stack trace**
N/A as no exception is thrown

**To Reproduce**
The following code using the postgresql driver should reproduce the issue:

```typescript
@Entity()
class MyEntity {
  @PrimaryKey({ autoincrement: true })
  id!: number;

  @Property({ type: 'json', nullable: true })
  someJson?: unknown;
}

const em1 = em.fork();

const entityIn = new MyEntity();
entityIn.someJson = '123';

em1.persist(entityIn);
await em1.flush();

const em2 = em.fork();

const entityOut = await em2.findOneOrFail({ id: entityIn.id });

// entityOut.someJson should be '123'
entityOut.someJson === '123';

// but it is 123
entityOut.someJson === 123;
```

**Expected behavior**
If a JSON property is set to a string, when populating from the db, it should be a string, even if that string happens to be valid JSON.

**Additional context**
This regression was introduced in https://github.com/mikro-orm/mikro-orm/pull/4194

The `parseJsonSafe` assumes that if the database returns a `JSON.parse()`-able string, it means that the database driver is not automatically parsing these values and that the ORM should. But that's not necessarily true. The JSON column could contain a JSON-encoded string that is itself JSON parseable.

**Versions**

| Dependency | Version |
| - | - |
| node | v16.13.1 |
| typescript | v4.9.5 |
| mikro-orm | v5.7.12 |
| your-driver | postgresql |

**Comment by B4nan** [comment]
Hmm I guess I will really need to disable the JSON processing on driver level completely (as that is the reason why we need this quirky json parsing), last time I tried it wasn't working as expected in some drivers, but there must be a way to do that...

**Comment by B4nan** [comment]
Closing as implemented in v6 via #4609
